src/algebra/free_monoid.lean

```diff
@@ -107,6 +107,16 @@ by { ext, simp }
 lemma hom_map_lift (g : M →* N) (f : α → M) (x : free_monoid α) : g (lift f x) = lift (g ∘ f) x :=
 monoid_hom.ext_iff.1 (comp_lift g f) x
 
+/-- Define a multiplicative action of `free_monoid α` on `β` from  -/
+@[to_additive] def mk_mul_action (f : α → β → β) : mul_action (free_monoid α) β :=
+{ smul := λ l b, list.foldr f b l,
+  one_smul := λ x, rfl,
+  mul_smul := λ xs ys b, list.foldr_append _ _ _ _ }
+
+@[simp, to_additive] lemma of_smul (f : α → function.End β) (x : α) (y : β) :
```

Comment:
Is there a reason why you used `α → function.End β` here and `α → β → β` above? I mean both are fine but why not use the same in both places?

Reply:
I wrote it using `function.End` first, then rewrote using `β → β` because we don't have the additive version of `function.End`. Forgot to fix it here.

```suggestion
@[simp, to_additive] lemma of_smul (f : α → β → β) (x : α) (y : β) :
```